batch/batch/front_end/front_end.py

```diff
@@ -351,14 +385,7 @@ async def _get_job_log_from_record(app, batch_id, job_id, record):
         tasks.append('output')
 
     if state == 'Running':
-        try:
-            resp = await request_retry_transient_errors(
-                client_session, 'GET', f'http://{ip_address}:5000/api/v1alpha/batches/{batch_id}/jobs/{job_id}/log'
-            )
-            return await resp.json()
-        except aiohttp.ClientResponseError:
-            log.exception(f'while getting log for {(batch_id, job_id)}')
-            return {task: 'ERROR: encountered a problem while fetching the log' for task in tasks}
+        return await handle_running_job(batch_id, job_id, tasks, client_session, ip_address)
```

Comment:
Ya I think I still find this structure quite confusing. I'd prefer not to pass in these `handle_running_job` and `handle_completed_job` pseudo-continuations and not have a function called `_get_resource_from_record`. I think we should be able to write the `get_job_log` and `get_job_resource_usage` functions without creating these handle callbacks. I'm not against breaking these pieces down into helper functions, but I don't think that these warrant passing functions in as arguments here as it requires jumping back and forth between function definitions when scanning the code linearly. Not sure if I'm making sense but happy to provide an example if I'm not.